File: go/muse/util/sequence_builder/sequence_builder.go
package sequence_builder

import (
	"muse/util/random_factory"
	"muse/util/sequences"
)

func PackIdentical(arr []int) {
	n := random_factory.GenInt()
	for i := range arr {
		arr[i] = n
	}
}

func PackIncreasing(arr []int) {
	if len(arr) == 0 {
		return
	}
	arr[0] = random_factory.GenIntN(1, 3)
	for i := 1; i < len(arr); i++ {
		arr[i] = arr[i-1] + random_factory.GenIntN(1, 3)
	}
}

func PackRandom(arr []int) {
	for i := range arr {
		arr[i] = random_factory.GenInt()
	}
}

func PackDecreasing(arr []int) {
	PackIncreasing(arr)
	sequences.Reverse(arr)
}

File: archive/java/muse/util/SequenceBuilder.java
package muse.util;

public final class SequenceBuilder {

  private SequenceBuilder() {}

  public static void packIncreasing(Integer[] arr) {
    if (arr.length == 0) {
      return;
    }
    arr[0] = RandomFactory.genIntN(1, 3);
    for (int i = 1, length = arr.length; i < length; i++) {
      arr[i] = arr[i - 1] + RandomFactory.genIntN(1, 3);
    }
  }

  public static void packRandom(Integer[] arr) {
    for (int i = 0, length = arr.length; i < length; i++) {
      arr[i] = RandomFactory.genInt();
    }
  }

  public static void packDecreasing(Integer[] arr) {
    packIncreasing(arr);
    Sequences.reverse(arr);
  }
}
